### ISSUE TITLE
"Please call preprocessVertexNeighbors() in a pre-process." in ttkScalarFieldSmoother

### ISSUE THREAD

**Issue by HollowSun0** [topology-tool-kit/ttk#231]
Hello,

I'm having weird results when using ttkScalarFieldSmoother on a closed PolyLine. The following message received is this one:

```
[Triangulation] VertexNeighborNumber query without pre-process!
[Triangulation] Please call preprocessVertexNeighbors() in a pre-process.
[Triangulation] VertexNeighborNumber query without pre-process!
[Triangulation] Please call preprocessVertexNeighbors() in a pre-process.
[...]
```

The error is reproducible with the following properties:
![image](https://user-images.githubusercontent.com/43864651/59459237-4a4dff00-8dea-11e9-8c96-0ff262a5e3f6.png)

... and by using the following input:
[problematicSlice.zip](https://github.com/topology-tool-kit/ttk/files/3287666/problematicSlice.zip)

I currently use a Windows 10 OS with MSVC 2017. The last ttk commit SHA we pulled is the following: 78fd8ca7b81d554b5166fe291c247ea15d874f44

**Comment by julien-tierny** [comment]
hi,

thanks for your message.

the issue #184 discussed a similar problem, which should have been fixed with commit ffd22f2.

can you try again with the latest version of the master branch?
also, can you share the corresponding data?

thanks,
julien
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny

**Comment by HollowSun0** [comment]
Thank you very much! Everything seems to work with that commit!

I can share the input and output data files corresponding to this issue:
[debug_issue_231.zip](https://github.com/topology-tool-kit/ttk/files/3291048/debug_issue_231.zip)

Have a good day,
Patrick Laurin
